packages/@webex/plugin-cc/src/services/task/index.ts

```diff
@@ -0,0 +1,90 @@
+import {LocalMicrophoneStream} from '@webex/calling';
+import {getErrorDetails} from '../core/Utils';
+import {LoginOption} from '../../types';
+import {CC_FILE} from '../../constants';
+import routingContact from './contact';
+import {ITask, TaskId, TaskResponse, TaskData} from './types';
+import WebCallingService from '../WebCallingService';
+
+export default class Task implements ITask {
+  private contact: ReturnType<typeof routingContact>;
+  private localAudioStream: LocalMicrophoneStream;
+  private webCallingService: WebCallingService;
+  public data: TaskData;
+
+  constructor(
+    contact: ReturnType<typeof routingContact>,
+    webCallingService: WebCallingService,
+    data: TaskData
+  ) {
+    this.contact = contact;
+    this.data = data;
+    this.webCallingService = webCallingService;
+  }
+
+  public updateTaskData = (newData: TaskData) => {
+    this.data = newData;
+
+    return this;
+  };
+
+  /**
+   * This is used for incoming task accept by agent.
+   *
+   * @param taskId - Unique Id to identify each task
+   *
+   * @returns Promise<TaskResponse>
+   * @throws Error
+   * @example
+   * ```typescript
+   * task.accept(taskId).then(()=>{}).catch(()=>{})
+   * ```
+   */
+  public async accept(taskId: TaskId): Promise<TaskResponse> {
```

Comment:
taskId should no longer be needed right?

Reply:
addressed